### ISSUE TITLE
Strange problem with constraint.py

### ISSUE THREAD

**Issue by maajeck** [perseas/Pyrseas#150]
Hi,

I'm running some databases and want to use Pyrseas to validate schema's. Though the most complex database gives me a strange problem. 

```
psycopg2.ProgrammingError: cannot cast type regclass to name
LINE 2: ...        CASE WHEN contypid = 0 THEN conrelid::regclass::name
```
The solution to this problem was explicitly left join pg_class vs. pg_type. The modified query in pyrseas/dbobject/constraint.py becomes: 
```
query = \
        """SELECT nspname AS schema,
                  CASE WHEN contypid = 0 THEN rel.relname
                       ELSE typ.typname END AS table,
                  conname AS name,
                  CASE WHEN contypid != 0 THEN 'd' ELSE '' END AS target,
                  contype AS type, conkey AS keycols,
                  condeferrable AS deferrable, condeferred AS deferred,
                  confrelid::regclass AS ref_table, confkey AS ref_cols,
                  consrc AS expression, confupdtype AS on_update,
                  confdeltype AS on_delete, confmatchtype AS match,
                  amname AS access_method, spcname AS tablespace,
                  indisclustered AS cluster, coninhcount > 0 AS inherited,
                  obj_description(c.oid, 'pg_constraint') AS description
           FROM pg_constraint c
                JOIN pg_namespace ON (connamespace = pg_namespace.oid)
                LEFT JOIN pg_class cl on (conname = relname)
                LEFT JOIN pg_index i ON (i.indexrelid = cl.oid)
                LEFT JOIN pg_tablespace t ON (cl.reltablespace = t.oid)
                LEFT JOIN pg_am on (relam = pg_am.oid)
		LEFT JOIN pg_class rel ON rel.oid=c.conrelid
		LEFT JOIN pg_type typ ON typ.oid=c.contypid
           WHERE (nspname != 'pg_catalog' AND nspname != 'information_schema')
           ORDER BY schema, 2, name"""
```

I don't know why psycopg2 complains, but this is a solution. It would be nice if someone can put this into the next version. 

Python 2.7.10, psycopg2 2.6.2, PyYAML 3.12, Pyrseas 0.7.2

Best regards,
Marco van Eck

**Comment by jmafc** [comment]
Hello Marco,

I believe this has been fixed in the master branch by @shirkey in change 84a365dd. His approach was to change the cast from `name` to `text`.  If you could verify that the change also works for your most complex database, that would be great.  The master branch will eventually be released as Pyrseas 0.8.

**Comment by maajeck** [comment]
Hi,

The change @shirkey made also works on my complex database. Sorry for not checking the master branch.
